foyer-common/src/code.rs

```diff
@@ -433,7 +395,7 @@ impl std::io::Read for ArcVecU8Cursor {
         let slice = self.inner.as_ref().as_slice();
         let len = std::cmp::min(slice.len() - self.pos, buf.len());
         buf.put_slice(&slice[self.pos..self.pos + len]);
-        self.pos -= len;
+        self.pos += len;
```

Comment:
Is it a bug introduced from previous PR ?

Reply:
Yep, this PR is part of a series to make each PR small. The previous one is not fully tested.